src/Symfony/Component/Serializer/Normalizer/ObjectNormalizer.php

```diff
@@ -152,4 +153,30 @@ protected function getAllowedAttributes($classOrObject, array $context, $attribu
 
         return $allowedAttributes;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getTypes(string $currentClass, string $attribute)
+    {
+        if (null !== $types = parent::getTypes($currentClass, $attribute)) {
+            return $types;
+        }
+
+        if (null !== $this->classDiscriminatorResolver && null !== $discriminatorMapping = $this->classDiscriminatorResolver->getMappingForClass($currentClass)) {
+            if ($discriminatorMapping->getTypeProperty() === $attribute) {
+                return array(
+                    new Type(Type::BUILTIN_TYPE_STRING),
+                );
+            }
+
+            foreach ($discriminatorMapping->getTypesMapping() as $mappedClass) {
+                if (null !== $types = parent::getTypes($mappedClass, $attribute)) {
+                    return $types;
+                }
+            }
+        }
+
+        return null;
```

Comment:
not needed, is it?

Reply:
imo it's better to say "returning `null`" rather than not returning anything (void)

Reply:
I'd argue it is worth being explicit as well.